SubEthaEdit-Mac/Source/TCMMillionMonkeys/TCMMMUserManager.m

```diff
@@ -51,16 +57,9 @@ - (id)init {
     return self;
 }
 
-- (void)dealloc {
-    [I_userRequestsByID release];
-    [I_usersByID release];
-    [I_me release];
-    [super dealloc];
-}
 
 - (void)setMe:(TCMMMUser *)aUser {
-    [I_me autorelease];
-     I_me = [aUser retain];
+     I_me = aUser;
```

Comment:
I'm not sure if i get the reason for `autoreleasing` the ivar before assigning a new one but from your explanation I assume the presence of structures like

```objective-c
// with ARC
@implementation Foo
-(NSObject *) foo { return I_foo; }
-(void) setFoo:(NSObject *)foo { I_foo = foo; }
@end

// without ARC
@implementation Bar
-(void)outer:(Foo *)o {
    NSObject *x = [o foo];
    [self inner:o];
    [x someSelector]; // x might be deallocated here
}

-(void)inner:(Foo *)o {
   [o setFoo:...]
}
@end
```

My preferred way would be to finish up arc conversion as the problem should not occur in ARC Code. This would also allow to replace all setters/getters with their `@synthesized` equivalent.

If thats not feasible for some reason, i'd go with a helper function like this

```objective-c 

// Without ARC
void TCMSendToAutoreleasePool(id o)  {
    [[o retain] autorelease];
}

// With ARC
-(void) setFoo:(NSObject *)foo { 
  TCMSendToAutoreleasePool(I_foo);
  I_foo = foo;
}


```

Reply:
Well, the reason before arc is the same as it was with collection classes. E.g. the classic example on the callsite:

```objc
NSString *oldValue = someDict[@"foo"]
someDict[@"foo"] = @"a brand new value"
// without arc, oldValue now is released
```

so if callsites are getting the current value, setting a new one, and working on with the current value, they potentially work with a released object. And that pattern is not as unlikely as one would like it to be.

And the places where there are specific retain/autoreleases might really be because they are needed because of the possibility to kill the callstack.